test/cypress/stubs/gateway-account-stubs.js

```diff
@@ -304,9 +313,19 @@ function postUpdateWorldpay3dsFlexCredentials (opts) {
   })
 }
 
-function patchUpdateCredentialsSuccess (gatewayAccountId, credentialId) {
+function patchUpdateCredentialsSuccess (gatewayAccountId, credentialId, patchOpts = {}) {
   const path = `/v1/api/accounts/${gatewayAccountId}/credentials/${credentialId}`
-  return stubBuilder('PATCH', path, 200)
+  return stubBuilder('PATCH', path, 200, {
+    request: [{
+      op: 'replace',
+      path: patchOpts.path,
+      value: patchOpts.value
+    }, {
+      op: 'replace',
+      path: 'last_updated_by_user_external_id',
+      value: patchOpts.userExternalId
+    }]
+  })
```

Comment:
I think this method signature has potential to be confusing. I'd prefer if a list of patchOpts were passed in, for e.g.
```
patchUpdateCredentialsSuccess(1, 123bc, [{path: 'my-path', value: 'my-value'}, {path: 'last_updated_by_user_external_id', value: 'a-user-external-id'}]

Reply:
Yeah this may not be perfect, perhaps `userExternalId` should instead be a separate parameter to pass in to the method instead. The `last_updated_by_user_external_id` replace op should always performed on this PATCH endpoint, so it probably doesn't need to be configured each time the stub is used, rather configured by default

Reply:
Yeah I think `userExternalId` should be a separate parameter. One annoying thing I find about method signatures with just `(opts)` ([example](https://github.com/alphagov/pay-selfservice/blob/master/test/cypress/stubs/agreement-stubs.js#L36) is that when calling the method from another file I have to look into the method implementation to find out what fields are available on the `opts`; this is something I'd like to see less of.

Reply:
`udpatedByUserExternalId` would be clearer than just `userExternalId` even.

Reply:
I do agree about the use of `opts`, it's very opaque about what's required. I've used `patchOpts` in a few places so I could add a JSdoc for it to make it more transparent

Reply:
Cool, whatever's easiest (adding a parameter or JSdoc)